doc/guides/dts/index.rst

```diff
@@ -346,44 +346,600 @@ Below is sample header content generated for this node, in
+    name.  The unit address portion of the prefix will be: ``400C0004_PTP``.
+    The full node prefix that combines the node compatible and unit address
+    will be: ``DT_NXP_KINETIS_PTP_400C0004_PTP``.
+
+    .. code-block:: none
+
+       soc {
+              temp1 {
+                      compatible = "nxp,kinetis-temperature";
+                      status = "okay";
+              };
+       };
+
+    For the ``temp1`` node above, since the node has no unit address the unit
+    address portion and the parent has no unit address we will utilize the
+    nodes name ``TEMP1`` as the unit address portion.
+    The full node prefix that combines the node compatible and unit address
+    will be: ``DT_NXP_KINETIS_TEMPERATURE_TEMP1``.
+
+``DT_INST_<instance-no.>_<compatible>``
+    The compatible string for the node together with an instance number.
+
+    The instance number is a unique index among all enabled
+    (``status = "okay"``) nodes that have a particular compatible string,
+    counting from zero. For example, if there are two enabled nodes that have
+    ``compatible = "foo,uart"``, then these node prefixes get generated:
+
+    .. code-block:: none
+
+       DT_INST_0_FOO_UART
+       DT_INST_1_FOO_UART
+
+    .. note::
+
+       The instance numbers in no way reflect any numbering scheme that
+       might exist for the SoC, in any node labels, or unit addresses for the
+       node.  The instance number is a pure count of enabled nodes with the
+       same compatible.  There is no guarantee that the same device
+       node is given the same instance number between builds.  The only
+       guarantee is that instance numbers will start at 0 and be contiguous
+       for the number of compatible instances that are enabled.
+
+``DT_ALIAS_<alias>``
+    Generated from the names of any properties in the ``/aliases`` node that
+    point to the node. See :ref:`dt-alias-chosen` for an overview.
+
+    For example, assume ``/aliases`` looks like this:
+
+    .. code-block:: none
+
+       aliases {
+               uart-1 = &uart1;
+       };
+
+    The alias name (``uart-1``) is converted to ``UART_1`` by uppercasing and
+    replacing non-alphanumeric characters with underscores, generating the node
+    prefix ``DT_ALIAS_UART_1`` for the ``&uart1`` node.
+
+    .. note::
+
+       Currently, an older deprecated ``DT_<compatible>_<alias>`` form is also
+       generated for aliases. For the example above, assuming the compatible
+       string for the ``&uart1`` node is ``"foo,uart"``, this gives
+       ``DT_FOO_UART_UART_1``.
+
+       Work is underway to replace this form with ``DT_ALIAS_*``.
+
+Property identifiers in macro names
+-----------------------------------
+
+Macros for particular properties on nodes have the form
+``DT_<node>_<property>``. ``<node>`` is one the node prefixes from above, and
+``<property>`` identifies the property.
+
+For all properties except ``reg`` (:ref:`documented below <reg_macros>`),
```

Comment:
I think this is still not corrected from my initial review:

https://github.com/zephyrproject-rtos/zephyr/pull/22782#discussion_r379151969

The truth is a lot more complex than this (over-)simplification.

Reply:
Fixed, reworked the wording there.